### PR TITLE
arduino-uno: Add HC-SR04 sensor example

### PR DESCRIPTION
**Pull request by Dajamante**
Added the example we worked on today.
Please let me know if there are too many / too few comments or if refactoring is necessary at this stage.

### CODE REVIEW

File: boards/arduino-uno/examples/uno-sensor.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+        // the trigger must be set to high under 10 µs as per the HC-SR04 datasheet
+        trig.set_high().void_unwrap();
+        delay.delay_us(10u16);
+        trig.set_low().void_unwrap();
+
+        while echo.is_low().void_unwrap() {
+            // exiting the loop if the timer has reached 200 ms.
+            // 0.2s/4µs = 50000
+            if timer1.tcnt1.read().bits()>=50000{
```
Comment (lights0123) [inline boards/arduino-uno/examples/uno-sensor.rs:54]:
```suggestion
            if timer1.tcnt1.read().bits() >= 50000 {
```

File: boards/arduino-uno/examples/uno-sensor.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                // jump to the beginning of the outer loop if no obstacle is detected
+                ufmt::uwriteln!(&mut serial, "Nothing was detected and jump to outer loop.\r").void_unwrap();
+                continue 'outer;
+            }
+        }
+        // Restarting the timer
+        timer1.tcnt1.write(|w| unsafe { w.bits(0) });
+
+        // Wait for the echo to get low again
+        while echo.is_high().void_unwrap(){}
```
Comment (lights0123) [inline boards/arduino-uno/examples/uno-sensor.rs:64]:
```suggestion
        while echo.is_high().void_unwrap() {}
```
`rustfmt` also collapses the `let mut pins = ...` at the beginning, which you may want to run

Comment (Dajamante) [reply boards/arduino-uno/examples/uno-sensor.rs]: Hi! Sorry I don't understand. What shall I change?

**Comment by Rahix** [comment]
Looks good! One minor thing: Please name the example `uno-hc-sr04` instead of `uno-sensor` so it's immeditately obvious what it is about.

**Comment by Rahix** [comment]
You forgot the `.rs` file-ending ;)

**Comment by Dajamante** [comment]
Sorry, fixed now!

**Pull request merged**